ci/changelog-summary.php

```diff
@@ -78,24 +78,27 @@ function debug( $arg ) {
  * @return int $merged_pr The PR object
  */
 function fetch_pr_merged_to_branch() {
-    $merged_pr = false;
     $missing_label = BRANCH === 'production' ? LABEL_DEPLOYED_PROD : LABEL_DEPLOYED_STAGING;
 
     $prs = curl_get( GITHUB_ENDPOINT . '/pulls?sort=created&direction=desc&state=closed&base=' . BRANCH );
+    echo "Fetching merged PRs - found " . sizeof( $prs ) . " candidates\n";
     foreach( $prs as $pr ) {
+        echo "Checking PR #{$pr['number']} for the merged PR...\n";
         if ( ! $pr['merged_at'] ?? '' ) {
-            // PR is closed but not merged, skip to next one.
+            echo "PR #{$pr['number']} is not merged, skipping.\n";
             continue;
         }
 
         $labels = array_map( fn ($label) => $label['name'], $pr['labels'] );
         if ( ! in_array( $missing_label, $labels ) ) {
-            // Look for PR with missing label.
-            $merged_pr = $pr;
+            echo "PR #{$pr['number']} - {$pr['title']} {$pr['html_url']} is it!\n";
+            return $pr;
```

Comment:
I think this will be the fix we need. Picks the most recent merge not the last one on the first page.

Reply:
Thank you!